src/lib.rs

```diff
@@ -954,6 +945,43 @@ impl<T: ArtifactOutput> ArtifactOutput for Project<T> {
     }
 }
 
+// Rebases the given path to the base directory lexically.
+//
+// The returned path from this function usually starts either with a normal component (e.g., `src`)
+// or a parent directory component (i.e., `..`), which is based on the base directory. Additionally,
+// this function converts the path into a UTF-8 string and replaces all separators with forward
+// slashes (`/`).
+//
+// The rebasing process is as follows:
+//
+// 1. Remove the leading components from the path that match the base components.
+// 2. Prepend `..` components to the path, equal in number to the remaining base components.
+fn rebase_path(base: impl AsRef<Path>, path: impl AsRef<Path>) -> PathBuf {
+    use path_slash::PathExt;
+
+    let base = base.as_ref();
+    let path = path.as_ref();
+
+    let mut new_path = path.components().collect::<std::collections::VecDeque<_>>();
+
+    for (i, (base_component, path_component)) in
+        base.components().zip(path.components()).enumerate()
+    {
+        if base_component == path_component {
+            new_path.pop_front();
+        } else {
+            let mut parent_dirs =
+                vec![std::path::Component::ParentDir; base.components().count() - i];
+            parent_dirs.extend(new_path);
+            new_path = parent_dirs.into();
+
+            break;
+        }
+    }
```

Comment:
I think we can solve this a bit differently
don't zip the two iterators but create the two iters separately and then iterate over the `path`
and continue as long as base and path components match

this way we don't need the vecdeque

Reply:
Hmm, I'm not sure I understand your suggestion.

If we compare the components of base and path at the same positions, it seems similar to using zip.

Could you please elaborate on the type of implementation you have in mind?

Reply:
something like

```rust
let mut components = Vec::new(); // perhaps we can even directly push components to a PathBuf?
let mut base_iter = base.components();
let mut path_iter = path.components();

while let Some(path) = path_iter.next() {
  if Some(path) != path_iter.next() {
     components.push(path);
     components.extend(path_iter);
  }
}


```

Reply:
Thank you for your explanation. I now understand your point. I've made the change and pushed it.